builtin/providers/azurerm/resource_arm_virtual_machine.go

```diff
@@ -1090,6 +1136,22 @@ func expandAzureRmVirtualMachineDataDisk(d *schema.ResourceData) ([]compute.Data
 	return data_disks, nil
 }
 
+func expandAzureRmVirtualMachineDiagnosticsProfile(d *schema.ResourceData) compute.DiagnosticsProfile {
+	diagnosticsProfiles := d.Get("diagnostics_profile").(*schema.Set).List()
```

Comment:
There are no checks here to see if we even have any config of that nature - this means that someone omitting this would get a panic when trying to dereference